src/collective/upload/static/main.js

```diff
@@ -186,32 +83,42 @@ $(document).ready(function() {
             });
 
             $(document).bind('drop', function (e) {
-                var url = $(e.originalEvent.dataTransfer.getData('text/html')).filter('img').attr('src');
-                if (url) {
-                    $.getImageData({
-                        url: url,
-                        server:'http://localhost:8080/Plone/@@jsonimageserializer?callback=?',
-                        success: function (img) {
-                            var canvas = document.createElement('canvas');
-                            canvas.width = img.width;
-                            canvas.height = img.height;
-                            if (canvas.getContext && canvas.toBlob) {
-                                canvas.getContext('2d').drawImage(img, 0, 0, img.width, img.height);
-                                canvas.toBlob(function (blob) {
-                                    $('#fileupload').fileupload('add', {files: [blob]});
-                                }, "image/jpeg");
-                            }
-                        },
-                        error: function(xhr, text_status){
-                            // Handle your error here
-                        }
-                    });
+              e.preventDefault();
+              var upload_image = function(img) {
+                var canvas = document.createElement('canvas');
+                canvas.width = img.width;
+                canvas.height = img.height;
+                canvas.getContext('2d').drawImage(img, 0, 0, img.width, img.height);
+                canvas.toBlob(function (blob) {
+                  $('#fileupload').fileupload('add', {files: [blob]});
+                }, "image/jpeg");
+              };
+              // drop multiple files from file system
+              var file, i, len, ref;
+              ref = e.originalEvent.dataTransfer.files;
+              for (i = 0, len = ref.length; i < len; i++) {
+                file = ref[i];
+                var reader = new FileReader();
+                reader.onload = function(e) {
+                  var img = new Image();
+                  img.onload = function() {
+                    upload_image(this);
+                  };
+                  img.src = e.target.result;
                 }
-                e.preventDefault();
+                reader.readAsBinaryString(file);
+              }
+              // drop one file from other site
+              var url = $('img', e.originalEvent.dataTransfer.getData('text/html')).attr('src');
+              if (url) {
+                $.getImageData({
+                  url: url,
+                  server:'http://localhost:8080/Plone/@@jsonimageserializer?callback=?',
```

Comment:
why to hard-code a server on `localhost:8080`? this is the cause of weird messages sometimes appearing in the folder contents view.